pkg/compactor/compactor_test.go

```diff
@@ -39,7 +39,9 @@ var _ = Describe("Running Compactor", func() {
 		embeddedEtcdQuotaBytes int64  = 8 * 1024 * 1024 * 1024
 		snapshotTimeout               = 30 * time.Second
 		defragTimeout                 = 30 * time.Second
-		needDefragmentation           = true
+		// keep SleepFor low so that tests are faster
+		sleepFor            = 10 * time.Nanosecond
```

Comment:
Please change this value to 0 seconds, since we don't really test the scraping of metrics in unit tests and rightly so. Unit tests should be very quick, and should not include any code that simply sleeps, unless the sleep is actually used for checking some functionality. Even if so, such tests should ideally be part of e2e tests.